com.ibm.streamsx.slack/com.ibm.streamsx.slack.service/SlackService.spl

```diff
@@ -0,0 +1,63 @@
+//
+// ****************************************************************************
+// * Copyright (C) 2017, International Business Machines Corporation          *
+// * All rights reserved.                                                     *
+// ****************************************************************************
+//
+
+/**
+ * A service which subscribes to a user-specified topic (in the parameters). 
+ * The incoming tuple should contain a jsonString. The jsonString, once converted, 
+ * should contain a message attribute with the text contents to send to Slack.
+ * 
+ * Input JSON Schema:
+ * {
+ *   message : string
+ * }
+ */
+
+namespace com.ibm.streamsx.slack.service;
+
+use com.ibm.streamsx.json::JSONToTuple ;
+use com.ibm.streamsx.slack::SendSlackMessage ;
+use com.ibm.streamsx.topology.topic::Subscribe ;
+
+public composite SlackService
```

Comment:
This isn't really a service as it can't be compiled and submitted as an application since it has required parameters. Ie. another SPL application is needed to invoke this composite.

Having the parameters default to submission parameters would be typically, maybe even those submission parameters having defaults.